cloudconfig/instancecfg/instancecfg.go

```diff
@@ -261,6 +257,9 @@ type SSHKeyPair struct {
 
 	// Public contains the public key in authorized_keys format.
 	Public string
+
```

Comment:
Above this we have `Private string` which is supposed to be a PEM encoded private key. Should we consider doing some maintenance work here to change the type to an actual PEM block so we have a strong type available while moving data round inside of Juju.

Reply:
This needs more work anyway to cleanup the duplicated SSHKeyPair structs